main_test.go

```diff
@@ -129,6 +135,61 @@ func TestPathFilter(t *testing.T) {
 	}
 }
 
+func TestLoadDefaultConfig(t *testing.T) {
+	originalConfig := *config
+	defer func() { config = &originalConfig }()
+
+	tmpdir, cleanup := setupTempDir(t)
+	defer cleanup()
+
+	mkFile(t, tmpdir, defaultConfigPath, `{"Deadline": "3m"}`)
+
+	app := kingpin.New("test-app", "")
+	app.Action(loadDefaultConfig)
+	setupFlags(app)
+
+	_, err := app.Parse([]string{})
+	require.NoError(t, err)
+	require.Equal(t, 3*time.Minute, config.Deadline.Duration())
+}
+
+func TestNoConfigFlag(t *testing.T) {
+	originalConfig := *config
+	defer func() { config = &originalConfig }()
+
+	tmpdir, cleanup := setupTempDir(t)
+	defer cleanup()
+
+	mkFile(t, tmpdir, defaultConfigPath, `{"Deadline": "3m"}`)
+
+	app := kingpin.New("test-app", "")
+	app.Action(loadDefaultConfig)
+	setupFlags(app)
+
+	_, err := app.Parse([]string{"--no-config"})
+	require.NoError(t, err)
+	require.Equal(t, 30*time.Second, config.Deadline.Duration())
+}
+
+func TestConfigFlagSkipsDefault(t *testing.T) {
+	originalConfig := *config
+	defer func() { config = &originalConfig }()
+
+	tmpdir, cleanup := setupTempDir(t)
+	defer cleanup()
+
+	mkFile(t, tmpdir, defaultConfigPath, `{"Deadline": "3m"}`)
+	mkFile(t, tmpdir, "test-config", `{}`)
```

Comment:
It would be good to set some value here so the test can check this config was loaded by checking the value.

Reply:
Done.